pkg/kubectl/apply/parse/list_element.go

```diff
@@ -127,12 +127,12 @@ func (v ElementBuildingVisitor) doMapList(meta apply.FieldMetaImpl, item *listIt
 		orderedKeys.UpsertRecorded(key, l)
 	}
 
-	for i, l := range orderedKeys.Items {
+	for _, l := range orderedKeys.Items {
 		var s proto.Schema
 		if item.Array != nil && item.Array.SubType != nil {
 			s = item.Array.SubType
 		}
-		subitem, err := v.getItem(s, fmt.Sprintf("%d", i), l.RawElementData)
+		subitem, err := v.getItem(s, fmt.Sprintf("%+v", l.KeyValue), l.RawElementData)
```

Comment:
Why is this change needed?  In general, not sure about using `%+v` for anything other than logging.  Could it at least be limited to the `PrimitiveListItem.Value` field, to exclude the embedded `RawElementData`?

Reply:
Aborted this change.